File: bench/tasks/18-path-parts/go/main.go
package main

import (
	"fmt"
	"path/filepath"
)

func main() {
	fmt.Println(filepath.ToSlash(filepath.Join("reports", "2026", "q1.json")))
	fmt.Println(filepath.Base("reports/2026/q1.json"))
	fmt.Println(filepath.ToSlash(filepath.Dir("reports/2026/q1.json")))
	fmt.Println(filepath.Ext("reports/2026/q1.json"))
}

File: bench/tasks/18-path-parts/rust/main.rs
use std::path::{Path, PathBuf};

fn main() {
    let joined: PathBuf = ["reports", "2026", "q1.json"].iter().collect();
    println!("{}", joined.display());
    let path = Path::new("reports/2026/q1.json");
    println!("{}", path.file_name().unwrap().to_string_lossy());
    println!("{}", path.parent().unwrap().display());
    println!(".{}", path.extension().unwrap().to_string_lossy());
}
